mxbluesky/devices/generic.py

```diff
@@ -36,17 +38,64 @@ def __init__(self, *args, **kwargs):
         self.cz = self.pz
         self.omega = self.o
 
-@unique
-class CryoStreamCmd(IntEnum):
-    START_RAMP = 1
-    STOP_RAMP = 0
+class Puck(Device):
+    status = Cpt(EpicsSignalRO, "-Sts")
 
+class Sector(Device):
+    A = Cpt(Puck, "A", name="A")
+    B = Cpt(Puck, "B", name="B")
+    C = Cpt(Puck, "C", name="C")
 
-class CryoStream(PVPositionerIsClose):
-    readback = Cpt(EpicsSignalRO, 'TEMP')
-    setpoint = Cpt(EpicsSignal, 'RTEMP')
-    actuate = Cpt(EpicsSignal, "RAMP.PROC")
-    actuate_value = CryoStreamCmd.START_RAMP
-    stop_signal = Cpt(EpicsSignal, "RAMP.PROC")
-    stop_value = CryoStreamCmd.STOP_RAMP
+class Dewar(Device):
+    rotation = Cpt(EpicsSignal, "{Dew:1-Ax:R}Virtual")
+    rotation_motor = Cpt(EpicsMotor, "{Dew:1-Ax:R}Mtr")
+    sectors = DDC(
+        {
+            f"sector_{i}": (Sector, f"{{Wago:1}}Puck{i}", {"name": "sector_{i}"})
+            for i in range(1, 9)
+        }
+    )
+    num_sectors = 8
+
+    def get_puck_status(self, puck_pos: str):
+        # puck_pos assumed to be of the form "7A"
+        sector = getattr(self.sectors, f"sector_{puck_pos[0]}")
+        puck: Puck = getattr(sector, puck_pos[1])
+        return puck.status.get()
+
+    def rotate(self, rotation_angle, absolute=True):
+        def check_value_sink(*, old_value, value, **kwargs):
+            "Return True when the movement is complete, False otherwise."
```

Comment:
I think these messages are backwards?

Reply:
I think I copied code you had provided 😄 
Should I switch it? It seems to be working as is. It's at the beamlines

Reply:
Yeah, it's confusing, just swap the comments between the two and I think it's fine.